### PR TITLE
append MASON_STDLIB_VERSION to binary if MASON_CXX_PACKAGE == true

### PR DESCRIPTION
**Pull request by mikemorris**
Prospective solution to https://github.com/mapbox/mason/issues/111, packages with stdlib ABI compatibility issues can opt-in to publishing/fetching compiler-specific versions.

- [x] Add documentation for the new variable(s) to README.md.

/cc @danpat @springmeyer

### CODE REVIEW

**Comment by mikemorris** [comment]
Anyone have a good regex handy for trimming a version like `4.2.1` or `4.8` to `major.minor` regardless of whether a patch number is present?

**Comment by jfirebaugh** [comment]
Please add documentation for the new variable(s) to README.md.

Needs @springmeyer :eyes: for the general approach.

File: mason.sh [hunk 1]
```diff
@@ -86,6 +86,19 @@ elif [ ${MASON_PLATFORM} = 'ios' ]; then
     export MASON_ISIM_CFLAGS="${MIN_SDK_VERSION_FLAG} -isysroot ${MASON_SDK_PATH} -arch i386 -arch x86_64"
 
 elif [ ${MASON_PLATFORM} = 'linux' ]; then
+    MASON_CXX=${CXX:-`c++ -v 2>&1`}
+
+    if [[ ${CXX} ]]; then
+        MASON_CXX_DUMPVERSION=`${CXX} -dumpversion`
+    else
+        MASON_CXX_DUMPVERSION=`c++ -dumpversion`
+    fi
```
Comment (kkaefer) [inline mason.sh]: Can we shorten this block to `${CXX:-c++} -dumpversion` and directly use it below?

File: mason.sh [hunk 2]
```diff
@@ -86,6 +86,19 @@ elif [ ${MASON_PLATFORM} = 'ios' ]; then
...
+
+    if [[ ${MASON_CXX} =~ "clang" ]]; then
+        export MASON_CXX_VERSION='-clang-'${MASON_CXX_DUMPVERSION}
```
Comment (kkaefer) [inline mason.sh]: Can we rename `MASON_CXX_VERSION` to `MASON_STL_VERSION`?

**Comment by mikemorris** [comment]
On hold per https://github.com/mapbox/mason/issues/111#issuecomment-144894538

**Pull request closed without merging**